### ISSUE TITLE
KSQL doesn't seems to be able to group by a map field

### ISSUE THREAD

**Issue by killfill** [confluentinc/ksql#606]
When trying to `group by` a `map` field, ksql tells this:

```sql
ksql> select backend_page['url'], count(*) from pageviews group by backend_page['url'];
Couldn't find field with name=PAGEVIEWS.BACKEND_PAGE['url'] in schema. fields=[org.apache.kafka.connect.data.Field@8da088c2, etc, etc]
```

One could 'flatten' the field, by creating a new stream out of the original one, extracting the backend_page['url'] field into its own `backend_page_url` and it works, but wondering its it would be possible to do that without an extra stream.

Thanks!

**Comment by rmoff** [comment]
@killfill do you have the sample input data for this and the `CREATE STREAM` statement, to help reproduce this issue? thanks.

**Comment by killfill** [comment]
Uups, funny.
I just tried to do the same thing and found this ticket.
Totally forgot about this ticket, sorry!

Sample Data:
```
$ echo '{"hello":"there","backend_page":{"url":"http://confluent.com/url1","host":"confluent.com","pathname":"/url1"}}' | kafkacat -b 10.10.0.27:31090 -t map -P
$ echo '{"hello":"there","backend_page":{"url":"http://confluent.com/url1","host":"confluent.com","pathname":"/url1"}}' | kafkacat -b 10.10.0.27:31090 -t map -P
$ echo '{"hello":"there2","backend_page":{"url":"http://confluent.com/url2","host":"confluent.com","pathname":"/url2"}}' | kafkacat -b 10.10.0.27:31090 -t map -P
```

KSQL:
```sql
ksql> SET 'auto.offset.reset' = 'earliest';
Successfully changed local property 'auto.offset.reset' to 'earliest'. Use the UNSET command to revert your change.
ksql> select * from a_stream;
1546003520705 | null | there2 | {host=confluent.com, url=http://confluent.com/url2, pathname=/url2}
1546003501481 | null | there | {host=confluent.com, url=http://confluent.com/url1, pathname=/url1}
1546003508736 | null | there | {host=confluent.com, url=http://confluent.com/url1, pathname=/url1}


^CQuery terminated
ksql> select hello, count(*) from a_stream group by hello;
there | 2
there2 | 1
^CQuery terminated
ksql> select backend_page['url'], count(*) from a_stream group by backend_page['url'];
name is null
```

I'm trying this with version 5.1.0 of ksql client and server.
As you can see the message now is 'name is null'

Maybe this is related to https://github.com/confluentinc/ksql/pull/2076 (?)

**Comment by apurvam** [comment]
Indeed, this should be fixed by #2076. Have you checked with KSQL 5.3?

**Comment by mjsax** [comment]
As there was not response, I assume it's fixed. Closing this ticket. Feel free to reopen if necessary.
